python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -407,6 +415,18 @@ def _update(self):
             )
             return
 
+        # Populate mapping of replica indices to nodes in that replica.
+        self.replicas_to_nodes.clear()
+        for node_id in self.non_terminated_nodes.worker_ids:
+            tags = self.provider.node_tags(node_id)
+            if TAG_RAY_REPLICA_INDEX in tags:
+                node_replica_index = tags[TAG_RAY_REPLICA_INDEX]
+                if node_replica_index in self.replicas_to_nodes:
+                    if node_id not in self.replicas_to_nodes[node_replica_index]:
```

Comment:
When will `node_id` in `self.replicas_to_nodes[node_replica_index]`? If we don't need this check, `if node_replica_index in self.replicas_to_nodes` also becomes redundant.

Reply:
Done in 47e46ac.